lib/templatesimplifier.cpp

```diff
@@ -1611,14 +1611,14 @@ void TemplateSimplifier::expandTemplate(
         Token * dstStart = dst->previous();
         bool isStatic = false;
         std::string scope;
-        Token * start;
-        Token * end;
+        const Token * start;
+        const Token * end;
         auto it = mTemplateForwardDeclarationsMap.find(dst);
         if (!isSpecialization && it != mTemplateForwardDeclarationsMap.end()) {
             dst = it->second;
             dstStart = dst->previous();
-            const Token * temp1 = dst->tokAt(1)->findClosingBracket();
-            const Token * temp2 = temp1->tokAt(getTemplateNamePosition(temp1));
+            Token * temp1 = dst->tokAt(1)->findClosingBracket();
```

Comment:
hmm I don't see why temp1 and temp2 can't be const here. why does the compiler complain?

Reply:
That is due to the incremental changes without actually introducing the proper signature/implementations yet, When it is properly fixed it might also require some `const` to be removed again.

I did not change the methods yet since they require the introduction of a few `const_cast` usages which I need to take a closer look at first. That will be done in the other PR.

Reply:
I restored the `const`.

Reply:
The `misc-const-correctness` clang-tidy check unfortunately doesn't support pointer yet otherwise I would have been able to run that before/after my changes to make sure they are all properly marked.